visualizer/IdealGraphVisualizer/SourceRepository/src/main/java/org/graalvm/visualizer/source/impl/ui/LocationOpener.java

```diff
@@ -62,25 +66,48 @@ private void openOrView(boolean focus) {
         if (toOpen == null) {
             return;
         }
-        int line = location.getLine();
+        int lineNumber = location.getLine();
+        int[] offsets = location.getOffsetsOrNull();
         EditorCookie cake = toOpen.getLookup().lookup(EditorCookie.class);
+        if (cake.getDocument() != null && offsets != null) {
+            lineNumber = NbDocument.findLineNumber(cake.getDocument(), offsets[0]);
+        }
+
+        Task task = cake.prepareDocument();
+        Line line = findLine(cake, lineNumber);
+
+        TaskListener[] whenShowing = new TaskListener[1];
+        whenShowing[0] = (ev) -> {
+            task.removeTaskListener(whenShowing[0]);
+            Mutex.EVENT.postReadRequest(() -> {
```

Comment:
after the document loads, please check that the offsets are still within document length; if not, fall back to just the line.

Reply:
Done in 9ee80e5